R/proxy.R

```diff
@@ -76,8 +76,12 @@ vec_data <- function(x) {
     x <- vec_set_attributes(x, list(names = names(x)))
   }
 
-  x
+  unset_s4(x)
```

Comment:
This doesn't seem necessary to me? The proxy of a S4 object should be a base vector.

Reply:
Can you add a comment that this is to support vector-like S4 objects, i.e. S4 objects that have a base vector type.

Reply:
Please add a test to make sure the input object is not unmarked without a copy.